instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -171,4 +214,12 @@ defmodule OpentelemetryEcto do
   end
 
   defp format_error(_), do: ""
+
+  defp db_system(db_system) when db_system in @db_systems, do: db_system
+  defp db_system(_), do: "other_sql"
+
+  defp db_system(nil, Ecto.Adapters.Postgres), do: "postgresql"
+  defp db_system(nil, Ecto.Adapters.MyXQL), do: "mysql"
+  defp db_system(nil, Ecto.Adapters.Tds), do: "mssql"
+  defp db_system(db_system, _), do: db_system(db_system)
```

Comment:
@tsloughter leaving a note here, as I said before, it will be prudent to ask `Ecto` folks to include a function under the adapter module that allow us to fetch the `engine` name ...

I am not sure how open they will be, but I think it is a fair thing to ask and see what happens.

"Ideally," this package wouldn't maintain the mapping